src/data_provider/src/extended_sources/groups/src/user_groups_linux.cpp

```diff
@@ -0,0 +1,133 @@
+/* Copyright (C) 2015, Wazuh Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <iostream>
+#include "user_groups_linux.hpp"
+#include "group_wrapper.hpp"
+#include "passwd_wrapper.hpp"
+#include "system_wrapper.hpp"
+
+UserGroupsProvider::UserGroupsProvider(std::shared_ptr<IGroupWrapperLinux> groupWrapper,
+                                       std::shared_ptr<IPasswdWrapperLinux> passwdWrapper,
+                                       std::shared_ptr<ISystemWrapper> sysWrapper)
+    : m_groupWrapper(std::move(groupWrapper))
+    , m_passwdWrapper(std::move(passwdWrapper))
+    , m_sysWrapper(std::move(sysWrapper))
+{
+}
+
+UserGroupsProvider::UserGroupsProvider()
+    : m_groupWrapper(std::make_shared<GroupWrapperLinux>())
+    , m_passwdWrapper(std::make_shared<PasswdWrapperLinux>())
+    , m_sysWrapper(std::make_shared<SystemWrapper>())
+{
+}
+
+nlohmann::json UserGroupsProvider::collect(const std::set<uid_t>& uids)
+{
+    nlohmann::json results = nlohmann::json::array();
+    struct passwd pwd;
+    struct passwd* pwd_results
+    {
+        nullptr
+    };
+
+    size_t bufsize = m_sysWrapper->sysconf(_SC_GETPW_R_SIZE_MAX);
+
+    if (bufsize > 16384)
+    {
+        /* Value was indeterminate */
+        bufsize = 16384; /* Should be more than enough */
```

Comment:
unify comments and / or replace by a const expre stating undefinition of a specific value.